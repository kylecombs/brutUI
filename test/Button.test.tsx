import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Button } from "../src";
import { FiArrowRight, FiStar } from "react-icons/fi";

afterEach(cleanup);

describe("Button", () => {
  it("Renders buttons correctly", () => {
    const { asFragment } = render(
      <>
        <Button variant="primary" disabled>
          Button CTA
        </Button>

        <Button variant="primary" size="sm">
          Button CTA
        </Button>

        <Button variant="secondary" size="md" LeadingIcon={<FiStar />}>
          Button CTA
        </Button>

        <Button
          variant="secondaryGray"
          size="lg"
          TrailingIcon={<FiArrowRight />}
        >
          Button CTA
        </Button>

        <Button variant="tertiary" size="xl" IconOnly={<FiArrowRight />} />

        <Button variant="tertiaryGray" size="2xl" IconOnly={<FiStar />} />
      </>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
