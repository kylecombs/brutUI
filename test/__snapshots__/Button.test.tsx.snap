// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders buttons correctly 1`] = `
<DocumentFragment>
  <button
    class="btn-base btn-md btn-primary-disabled cursor-not-allowed"
  >
    Button CTA
  </button>
  <button
    class="btn-base btn-sm btn-primary btn-primary-hover btn-primary-focus shadow-grayDark"
  >
    Button CTA
  </button>
  <button
    class="btn-base btn-md btn-secondary btn-secondary-hover btn-secondary-focus shadow-grayDark"
  >
    <svg
      class="mr-2"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    Button CTA
  </button>
  <button
    class="btn-base btn-lg btn-secondaryGray btn-secondaryGray-hover btn-secondaryGray-focus"
  >
    Button CTA
    <svg
      class="ml-2"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </button>
  <button
    class="btn-base btn-icon-xl justify-center btn-tertiary btn-tertiary-hover"
  >
    <svg
      fill="none"
      height="20"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="5"
        x2="19"
        y1="12"
        y2="12"
      />
      <polyline
        points="12 5 19 12 12 19"
      />
    </svg>
  </button>
  <button
    class="btn-base btn-icon-2xl justify-center btn-tertiaryGray btn-tertiaryGray-hover"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
  </button>
</DocumentFragment>
`;
